src/app/d2-loadout-builder/locked-armor/popup/LockablePopup.tsx

```diff
@@ -129,14 +129,11 @@ export default class LockablePopup extends React.Component<Props, State> {
         )}
         {tabSelected === 'perks' && (
           <LockablePerks
-            {...{
-              perks,
-              locked,
-              hoveredPerk,
-              onPerkHover: this.onPerkHover,
-              reset: this.reset,
-              toggleLockedPerk: this.toggleLockedPerk
-            }}
+            perks={perks}
```

Comment:
any reason to prefer one style over another?

Reply:
The spelled-out props are a bit more efficient, but more importantly the spread version confuses TypeScript and no longer gets properly typechecked, so you won’t get notified of problems. It’s kind of annoying to spell them all out but I think it’s better overall.